src/main/java/Deadline.java

```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task {
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : " ");
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
```

Comment:
To me it seems a little unintuitive as to what D might mean if it's my first time seeing the code, perhaps you could make it an attribute of the Deadline class?